static/src/js/containers/MapContainer/MapContainer.tsx

```diff
@@ -473,24 +470,14 @@ export const MapContainer: React.FC<MapContainerProps> = (props) => {
           : deemphisizedGranuleStyle(index)
       }
 
-      let granuleGibsData: Partial<GibsData> | undefined
-
-      if (gibsTag) {
-        const gibsTime = gibsData.layerPeriod?.toLowerCase() === 'subdaily' ? timeStart : timeStart.substring(0, 10)
-
-        granuleGibsData = {
-          ...gibsData,
-          opacity: shouldDrawRegularStyle ? 1 : 0.5,
-          time: gibsTime,
-          url: `https://gibs-{a-c}.earthdata.nasa.gov/wmts/${projection}/best/wmts.cgi?TIME=${gibsTime}`
-        }
-      }
+      // Create GIBS data for each available GIBS layer using the new function
+      const granuleGibsData = buildGibsData(layersForProjection, projection, timeStart)
```

Comment:
`imageryLayers` is iterating over `layersForProjection`, and `buildGibsData` is also iterating over the same array. Avoid doing the same work twice

Reply:
We'll write a new ticket to improve this